### PR TITLE
Feature/add azure driver

### PR DESCRIPTION
**Pull request by dzmitryastrouski**

### CODE REVIEW

File: AZURE/create.yml [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+            key_data: "{{ keypair.public_key }}"
+
+    # Get unique idempotent name for VM, in Azure VM name == ID
+    # for Windows VM name cannot be longer than 15 characters
+    # VM name sample "tst-c03a89e9-vm"
+    - name: Set unique VM name dicts
+      set_fact:
+        unique_vm_name: >-
+          {{ unique_vm_name | default({})
+            | combine({ item.name: 'tst-' + ((item.name + '-' + ci_branch_name) | to_uuid)[:8] }) }}
```
Comment (tgadiev) [inline AZURE/create.yml:96]: Why only 8 characters length?

File: AZURE/create.yml [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+        resource_group: "{{ item.item.item.az_resource_group_name }}"
+        tags:
+          Instance: "{{ item.item.item.name }}"
+          CI_job: "{{ item.item.item.name }}-{{ ci_branch_name }}"
+          Name: "{{ unique_vm_name[item.item.item.name] }}"
+          Created_time: "{{ lookup('pipe','date +%Y-%m-%d') }}"
+          Managed_by: "Molecule"
+      loop: "{{ nic_create.results }}"
+      register: ip_tags
+      async: 7200
```
Comment (tgadiev) [inline AZURE/create.yml:147]: Is this task running so long we need async?

**Comment by pavelpikta** [comment]
Before merge need:
* update https://github.com/lean-delivery/ansible-development-kit
* prepare azure gitlab-runner

File: AZURE/create.yml (hunk 1 above)
Comment (dzmitryastrouski) [reply AZURE/create.yml:96]: Please see comments above
# for Windows VM name cannot be longer than 15 characters
# VM name sample "tst-c03a89e9-vm"
tst prefix for identifying and grouping VMs visually
vm postfix is recommended by Azure for identifying VM among few other VM parts as disc, NIC, IP address with the same name

File: AZURE/create.yml (hunk 2 above)
Comment (dzmitryastrouski) [reply AZURE/create.yml:147]: Yes it is. Unfortunately, every operation with resource takes time in Azure, even adding tags to the resource :(

File: AZURE/create.yml [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+    # For Windows VMs only
+    - name: create Azure VM extension to enable HTTPS WinRM listener
+      azure_rm_virtualmachineextension:
+        name: winrm-extension
+        location: "{{ item.item.item.az_location }}"
+        resource_group: "{{ item.item.item.az_resource_group_name }}"
+        virtual_machine_name: "{{ item.ansible_facts.azure_vm.name  }}"
+        publisher: Microsoft.Compute
+        virtual_machine_extension_type: CustomScriptExtension
+        type_handler_version: 1.9
```
Comment (tgadiev) [inline AZURE/create.yml]:
```suggestion
        type_handler_version: '1.9'
```

File: AZURE/create.yml [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+        settings: >-
+          {'commandToExecute': 'powershell -ExecutionPolicy Unrestricted -File {{ winrm_script_name }} -EnableCredSSP',
+            'fileUris': ['{{ winrm_script }}']}
+        auto_upgrade_minor_version: true
+      loop: "{{ instance_create.results }}"
```
Comment (tgadiev) [inline AZURE/create.yml:225]: Need to add async execution according to elapsed time.

**Review by tgadiev — CHANGES_REQUESTED** [review]

File: AZURE/create.yml (hunk 4 above)
Comment (dzmitryastrouski) [reply AZURE/create.yml:225]: Done.

File: AZURE/create.yml [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+  tasks:
+    - name: debug CI_COMMIT_REF_NAME
+      debug:
+        var: ci_branch_name
+
+    - name: Generate keypair
+      openssh_keypair:
+        # todo: must be protected!!!
+        path: /tmp/id_ssh_rsa
+        force: True
```
Comment (tgadiev) [inline AZURE/create.yml]:
```suggestion
        force: true
```

File: AZURE/create.yml [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+        name: "{{ unique_vm_name[item.name] }}-nic"
+        resource_group: "{{ item.az_resource_group_name }}"
+        virtual_network: "{{ item.az_virtual_network_name }}"
+        subnet_name: "{{ item.az_subnet_name }}"
+        location: "{{ item.az_location }}"
+        create_with_security_group: false
+        security_group: "{{ item.az_network_security_group }}"
+        ip_configurations:
+          - name: ipconfig1
+            primary: True
```
Comment (tgadiev) [inline AZURE/create.yml]:
```suggestion
            primary: true
```

**Review by tgadiev — CHANGES_REQUESTED** [review]

**Review by pavelpikta — APPROVED** [review]

File: AZURE/create.yml (hunk 5 above)
Comment (dzmitryastrouski) [reply AZURE/create.yml]: Fixed. Commit #fd7fd2

File: AZURE/create.yml (hunk 6 above)
Comment (dzmitryastrouski) [reply AZURE/create.yml]: Fixed. Commit #fd7fd2

**Review by tgadiev — APPROVED** [review]

**Pull request merged**